reana_db/models.py

```diff
@@ -1151,3 +1155,28 @@ class QuotaHealth(enum.Enum):
     healthy = 0
     warning = 1
     critical = 2
+
+
+class AccessType(enum.Enum):
+    """Enumeration of access types."""
+
+    read = 0
+
+
+class UserWorkflow(Base):
+    """UserWorkflow table."""
+
+    __tablename__ = "user_workflow"
+    __table_args__ = {"schema": "__reana"}
+
+    workflow_id = Column(UUIDType, ForeignKey("__reana.workflow.id_"), primary_key=True)
+    user_id = Column(UUIDType, ForeignKey("__reana.user_.id_"), primary_key=True)
+    access_type = Column(Enum(AccessType), default=AccessType.read, nullable=False)
+    message = Column(Text)
+    valid_until = Column(DateTime)
+
+    user = relationship("User", backref="user_workflows")
```

Comment:
(PS: also the UserWorkflow table name could be changed to something less generic, up to you!)

Reply:
With my recent changes, the User table now has:

1. `user.workflows`
2. `user.workflows_shared_with_me`

As for 3, I decided to remove it since it is not being used. I did not change the `UserWorkflow` table name because I think the current name is fine, it follows the standard many to many relationship naming convention

The changes were applied to the following superseding PR: https://github.com/reanahub/reana-db/pull/216